### ISSUE TITLE
[Feature Request] Swipe only through search results

### ISSUE THREAD

**Issue by floppyD** [SimpleMobileTools/Simple-Gallery#1279]
Like when I search ".webm", the search will return all files I have with this extension, but after opening a file and swiping, it doesn't swipe to the next search result, which is the desired behavior, instead it swipes to the next file on the folder. Also, after swiping and pressing back, the search results disappear, returning to the folder view.

**Comment by privatmamtora** [comment]
Can this please be split into 2 issues?

1. [Feature Request] Swipe through search results
2. [Bug] After selecting an item from search and pressing back, the folder when the search began is shown. If proper order is supposed to be maintained, then back should really be "back"
Correct order: "Folder View" > Select "Search" > "Search Results" > Select "item" > Press "Back" > **"Search Results" > Press "Back" >** "Folder View"
Current order: "Folder View" > Select "Search" > "Search Results" > Select "item" > Press "Back" > "Folder View"

**Comment by KnIfER** [comment]
As for the bug, I have created an PR !  
https://github.com/SimpleMobileTools/Simple-Gallery/pull/1510

**Comment by Jeder321** [comment]
Anything changed since last time? This feature is rather crucial to me because I work on big folders with lots of files I distinguish by using a common word, and so far I found one file explorer that lets me view search results like a gallery, however I'd prefer having it in one app.

**Comment by tibbi** [comment]
not really, nothing changed. It would need a new setting item as swiping through search results definitely isnt always the expected behaviour.

**Comment by Eskuero** [comment]
+1 for this feature, many times i give files names as tags and would really like to be able to swipe between then search results rather than the whole folder.

**Comment by saefthe** [comment]
مرحبا
شكرا لكم لا اقبل بحذف اي ملف او مقطع او صور او ملف صوت

Saef

في الثلاثاء، ١٩ تموز ٢٠٢٢, ٢:٥١ ص Óliver García Albertos <
***@***.***> كتب:

**Comment by privatmamtora** [comment]
@tibbi Depends on how you phrase it, it might actually be expected.
Currently when a user is "viewing" an image, they can swipe left and right to go to the previous or next image from the "Previous View" (Normally a Folder view)
Why shouldn't the same behavior not happen when the previous view is a "Search Results" view?

The fact that swiping an image from search results, shows "other" images, not in the search results, would be the "unexpected behavior" (Whenever it happens to me, I go "wait this wasn't in the results")

An example scenario is if I search for "Astrophotography", view an image, start swiping and suddenly I have random non Astrophotography images showing up from a folder.
